### DISCUSSION TITLE
wire:model to array with value in different type causes strange sync

### DISCUSSION THREAD

**Discussion by Yinci** [livewire/livewire#3435] [7. Bugs (archived)]
I have a set of rows with checkboxes that are bound to a single array, `$selected`. When selecting these checkboxes, the `$selected` property fills with the IDs of these rows, but in string format, since that is how HTML passes it. 

At the top, there is a 'select all' checkbox, which selects all rows. This is done by grabbing the IDs from a query, and setting `$selected` to these IDs. This is where things go wrong: All checkboxes are now checked, because the value is reasoned by Livewire as being selected (which they are, of course). However, when clicking a checkbox to unselect it, it tries to remove a string value from an array of integers. This value doesn't exist in this type, and therefore is not removed from the array. When then selecting the checkbox again, it appends the string value of the ID to the array. The end result: 
```php
[ 
    0 => 1
    1 => 2
    2 => '2'
];
```

Is there a way we could losely type values within an array when they are linked? Right now I have a map on the values returned to force them into string format. 

Laravel: 8.37
Livewire: 2.4.2

**Comment by joshhanley** [comment]
@Yinci yeah I tend to cast all my values to strings when I load the array initially, then convert to int when I'm trying to persist them.

That way everything is consistent and LW won't double up on 2 and '2'.

I think the main issue is anything coming from the html request is actually a string, so if you needed them to be ints you'd have to cast them in a hook like `updating`.

Hope this helps!
